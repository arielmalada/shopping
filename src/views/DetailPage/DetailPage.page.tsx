//#region IMPORTS

import { Button, Card, CardBody, Typography } from "@material-tailwind/react";
import React, { useContext } from "react";
import {
  ProductContext,
  ProductProvider,
} from "../../contexts/ProductProvider";
//#endregion

//#region SECONDARY COMPONENT
const Pictures: React.FC = () => {
  const productValue = useContext(ProductContext);
  return (
    <img
      src={productValue?.thumbnail}
      alt={productValue?.title || "empty image"}
    />
  );
};

const Data: React.FC = () => {
  const productValue = useContext(ProductContext);
  const isPriceReduced = (productValue?.discountPercentage || 0) > 0;
  const finalPrice =
    isPriceReduced && productValue
      ? productValue.price -
        Math.round(productValue.price * productValue.discountPercentage * 0.01)
      : productValue?.price;
  return (
    <div>
      <Typography variant="h2">{productValue?.title}</Typography>
      <div className="flex space-x-2">
        {isPriceReduced && (
          <div className="text-xs text-red-700 ">
            <div>Deal</div>
            <div className="text-sm">{productValue?.discountPercentage} %</div>
          </div>
        )}
        <div>$ {finalPrice}</div>
      </div>
      {isPriceReduced && (
        <div className="flex space-x-1 text-xs">
          <span>was</span>
          <span className="line-through">$ {productValue?.price}</span>
        </div>
      )}
    </div>
  );
};

const CheckoutSection: React.FC = () => {
  const productValue = useContext(ProductContext);
  const isPriceReduced = (productValue?.discountPercentage || 0) > 0;
  const finalPrice =
    isPriceReduced && productValue
      ? productValue.price -
        Math.round(productValue.price * productValue.discountPercentage * 0.01)
      : productValue?.price;
  return (
    <Card>
      <CardBody className="space-y-2">
        <Typography variant="h4">$ {finalPrice}</Typography>
        <Button>Add to Cart</Button>
      </CardBody>
    </Card>
  );
};
//#endregion

//#region MAIN COMPONENT
const DetailPage: React.FC = () => {
  return (
    <ProductProvider>
      <div>
        <section className="flex items-center justify-center space-x-10 xl:max-w-5xl">
          <Pictures />
          <Data />
          <CheckoutSection />
        </section>
        <section>related</section>
      </div>
    </ProductProvider>
  );
};
//#endregion

export default DetailPage;
