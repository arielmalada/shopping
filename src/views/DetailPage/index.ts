export {default} from './DetailPage.page'