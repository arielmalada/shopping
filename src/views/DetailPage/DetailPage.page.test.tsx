import React from 'react';
import renderer from 'react-test-renderer';
import DetailPage from './DetailPage.page';

describe('Detail Page', () => {
  it('should be able to render without crashing', () => {
    const tree = renderer.create(<DetailPage />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});