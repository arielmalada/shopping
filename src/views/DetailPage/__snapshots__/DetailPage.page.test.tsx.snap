// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Page should be able to render without crashing 1`] = `
<div>
  <section
    className="flex items-center justify-center space-x-10 xl:max-w-5xl"
  >
    <img
      alt="empty image"
    />
    <div>
      <h2
        className="block antialiased tracking-normal font-sans text-4xl font-semibold leading-[1.3] text-inherit"
      />
      <div
        className="flex space-x-2"
      >
        <div>
          $ 
        </div>
      </div>
    </div>
    <div
      className="relative flex flex-col bg-clip-border rounded-xl bg-white text-gray-700 shadow-md"
    >
      <div
        className="p-6 space-y-2"
      >
        <h4
          className="block antialiased tracking-normal font-sans text-2xl font-semibold leading-snug text-inherit"
        >
          $ 
        </h4>
        <button
          className="middle none font-sans font-bold center uppercase transition-all disabled:opacity-50 disabled:shadow-none disabled:pointer-events-none text-xs py-3 px-6 rounded-lg bg-blue-500 text-white shadow-md shadow-blue-500/20 hover:shadow-lg hover:shadow-blue-500/40 focus:opacity-[0.85] focus:shadow-none active:opacity-[0.85] active:shadow-none"
          onMouseDown={[Function]}
          type="button"
        >
          Add to Cart
        </button>
      </div>
    </div>
  </section>
  <section>
    related
  </section>
</div>
`;
